### PR TITLE
Improve ONNX importer API documentation

### PR DESCRIPTION
**Pull request by mbencer**
- cleaned included header files (left only needed)
- moved `register_operator` to separate file - `onnx_utils.cpp` (as a bit confused part of main API) - to discuss
- improved functions documentation (content, style, added info about error handling) - please comment
- changed some params name

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 1]
```diff
@@ -18,69 +18,65 @@
...
-                               Operator fn);
-
         /// \brief      Return the set of names of supported operators.
         ///
-        /// \param[in]  version  The requested version of ONNX operators set.
-        /// \param[in]  domain   The requested domain the operators where registered for.
-        ///
-        /// \return     The set containing names of supported operators.
+        /// \param[in]  version  The requested version of ONNX operators set
+        /// \param[in]  domain   The requested domain the operators where registered for
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: The description got garbled here somewhere; can you clarify this?

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 2]
```diff
@@ -18,69 +18,65 @@
...
         NGRAPH_API
-        std::shared_ptr<Function> import_onnx_model(std::istream& sin);
+        std::shared_ptr<Function> import_onnx_model(std::istream& stream);
 
-        /// \brief Convert an ONNX model to nGraph functions
-        /// The function translated serialized ONNX model to nGraph functions. The ONNX model
-        /// is read from ONNX file.
-        /// \param filename  file name (relative or absolute path name)
-        /// \return The function returns a nGraph function representing single output from graph.
+        /// \brief     Import and convert an ONNX model from file to nGraph function.
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: to an nGraph function?  to an nGraph `Function` namespace?

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 3]
```diff
@@ -18,69 +18,65 @@
...
+        ///            The function translated serialized ONNX model to nGraph functions.
+        ///            The ONNX model is read from ONNX file.
+        ///
+        /// \note      If parsing file fails or ONNX model contains not supported ops
+        ///            ngraph_error exception can be thrown.
+        ///
+        /// \param[in] file_path  The path to file containing ONNX model (relative or absolute)
+        ///
+        /// \return    The function returns a nGraph function representing single output from graph
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: an nGraph function representing ...

Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: I write "The function returns a nGraph Function which represents single output from the created graph", is OK?

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 2 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: I write "to an nGraph Function representation", is OK?

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 1 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: I rewrite it. Is it acceptable now?

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 4]
```diff
@@ -46,35 +46,33 @@ namespace ngraph
...
+        /// \brief      Import and convert an ONNX model from stream to an nGraph Function representation.
         ///             The serialized ONNX model is read from input stream.
         ///
         /// \note       If parsing stream fails or ONNX model contains not supported ops
         ///             ngraph_error exception can be thrown.
         ///
         /// \param[in]  stream      The input stream (e.g. file stream, memory stream, etc)
         ///
-        /// \return     The function returns a nGraph function representing single output from graph
+        /// \return     The function returns a nGraph Function which represents single output from the created graph
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: "an nGraph"

precede vowel sounds with "an" not "a"

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 5]
```diff
@@ -46,35 +46,33 @@ namespace ngraph
...
+        /// \brief     Import and convert an ONNX model from file to an nGraph Function representation.
         ///            The ONNX model is read from ONNX file.
         ///
         /// \note      If parsing file fails or ONNX model contains not supported ops
         ///            ngraph_error exception can be thrown.
         ///
         /// \param[in] file_path  The path to file containing ONNX model (relative or absolute)
         ///
-        /// \return    The function returns a nGraph function representing single output from graph
+        /// \return    The function returns a nGraph Function which represents single output from the created graph
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: Same comment as above

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 6]
```diff
@@ -29,10 +29,10 @@ namespace ngraph
 {
     namespace onnx_import
     {
-        /// \brief      Return the set of names of supported operators.
+        /// \brief      Return the set of names of supported operators which are supported.
```
Comment (indie) [inline src/ngraph/frontend/onnx_import/onnx.hpp]: Accidental redundancy introduced here?

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 2 above)
Comment (indie) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: Yes, that's the right way.

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 1 above)
Comment (indie) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: LGTM

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 6 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: right, `which are supported` removed

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 4 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp (hunk 5 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 7]
```diff
@@ -18,37 +18,23 @@
...
-        /// \param domain  domain the operator belongs to,
-        /// \param fn       function providing the implementation of the operator.
-        NGRAPH_API
-        void register_operator(const std::string& name,
-                               std::int64_t version,
-                               const std::string& domain,
-                               Operator fn);
-
-        /// \brief      Return the set of names of supported operators.
+        /// \brief      Returns the set of names of supported operators.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 8]
```diff
@@ -18,37 +18,23 @@
...
         ///
-        /// \param[in]  version  The requested version of ONNX operators set.
-        /// \param[in]  domain   The requested domain the operators where registered for.
+        /// \param[in]  version  Determines set version of operators which are returned.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 9]
```diff
@@ -18,37 +18,23 @@
...
+        /// \param[in]  domain   Determines domain of operators which are returned.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 10]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
         /// \param[in]  op_name  The ONNX operator name.
         /// \param[in]  version  The ONNX operator set version.
         /// \param[in]  domain   The domain the ONNX operator is registered to.
+        ///                      If not set default domain "ai.onnx" is used.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 11]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
         ///
         /// \return     True if operator is supported, False otherwise.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 12]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
         bool is_operator_supported(const std::string& op_name,
                                    std::int64_t version,
                                    const std::string& domain = "ai.onnx");
 
-        /// \brief Convert an ONNX model to nGraph function
-        /// The function translated serialized ONNX model to nGraph function. The serialized
-        /// ONNX model is read from input stream.
-        /// \param sin       input stream (e.g. file stream, memory stream, etc)
-        /// \return The function returns a nGraph function representing single output from graph.
+        /// \brief      Imports and converts an ONNX model from stream
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 13]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///             to an nGraph Function representation.
+        ///             The serialized ONNX model is read from input stream.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done, right, I added `serialized` to line above

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 14]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///
+        /// \note       If parsing stream fails or ONNX model contains not supported ops
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 15]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///             ngraph_error exception can be thrown.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 16]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///
+        /// \param[in]  stream      The input stream (e.g. file stream, memory stream, etc).
+        ///
+        /// \return     The function returns an nGraph Function which represents single output
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 17]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
         NGRAPH_API
-        std::shared_ptr<Function> import_onnx_model(std::istream& sin);
+        std::shared_ptr<Function> import_onnx_model(std::istream& stream);
 
-        /// \brief Convert an ONNX model to nGraph functions
-        /// The function translated serialized ONNX model to nGraph functions. The ONNX model
-        /// is read from ONNX file.
-        /// \param filename  file name (relative or absolute path name)
-        /// \return The function returns a nGraph function representing single output from graph.
+        /// \brief     Imports and converts an ONNX model from file
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 18]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///            to an nGraph Function representation.
+        ///            The ONNX model is read from ONNX file.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 19]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///
+        /// \note      If parsing file fails or ONNX model contains not supported ops
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 20]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///            ngraph_error exception can be thrown.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 21]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///
+        /// \param[in] file_path  The path to file containing ONNX model (relative or absolute).
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 22]
```diff
@@ -58,29 +44,41 @@ namespace ngraph
...
+        ///
+        /// \return    The function returns an nGraph Function which represents single output
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx_utils.hpp [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        /// \brief      Registers ONNX custom operator.
+        ///             The function performs the registration of external ONNX operator.
+        ///             which is not part of ONNX importer.
+        ///
+        /// \note       The operator shall be registered before calling
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/onnx_utils.hpp]:
```suggestion
        /// \note       The operator must be registered before calling
```

File: src/ngraph/frontend/onnx_import/onnx_utils.hpp [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+        ///             "import_onnx_model" functions.
+        ///
+        /// \param      name      The ONNX operator name.
+        /// \param      version   The ONNX operator set version.
+        /// \param      domain    The domain the ONNX operator is registered to.
+        /// \param      fn        The function providing the implementation of the operator.
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/onnx_utils.hpp]: Let's add a short description of the `Operator` function signature.

Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx_utils.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx_utils.hpp (hunk 23 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx_utils.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 25]
```diff
@@ -18,69 +18,68 @@
...
 
-#include "core/operator_set.hpp"
 #include "ngraph/function.hpp"
+#include "ngraph/ngraph_visibility.hpp"
 
+/// \brief              Top level NGraph namespace.
 namespace ngraph
 {
+    /// \brief          ONNX importer features namespace.
+    ///                 Elements belongs to the namespace allow to use ONNX format.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

File: src/ngraph/frontend/onnx_import/onnx.hpp [hunk 26]
```diff
@@ -18,69 +18,68 @@
...
 #include <iostream>
+#include <memory>
 #include <set>
 #include <string>
 
-#include "core/operator_set.hpp"
 #include "ngraph/function.hpp"
+#include "ngraph/ngraph_visibility.hpp"
 
+/// \brief              Top level NGraph namespace.
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/onnx.hpp]: done

**Pull request merged**